src/test/java/net/snowflake/ingest/streaming/internal/RowBufferTest.java

```diff
@@ -1498,4 +1499,71 @@ private void testE2EArrayHelper(OpenChannelRequest.OnErrorOption onErrorOption)
     ChannelData<?> result = innerBuffer.flush("my_snowpipe_streaming.bdec");
     Assert.assertEquals(5, result.getRowCount());
   }
+
+  @Test
+  public void testOnErrorAbortRowsWithError() {
+    AbstractRowBuffer<?> innerBufferOnErrorContinue =
+        createTestBuffer(OpenChannelRequest.OnErrorOption.CONTINUE);
+    AbstractRowBuffer<?> innerBufferOnErrorAbort =
+        createTestBuffer(OpenChannelRequest.OnErrorOption.ABORT);
+
+    ColumnMetadata colChar = new ColumnMetadata();
+    colChar.setName("COLCHAR");
+    colChar.setPhysicalType("LOB");
+    colChar.setNullable(true);
+    colChar.setLogicalType("TEXT");
+    colChar.setByteLength(14);
+    colChar.setLength(11);
+    colChar.setScale(0);
+
+    innerBufferOnErrorContinue.setupSchema(Collections.singletonList(colChar));
+    innerBufferOnErrorAbort.setupSchema(Collections.singletonList(colChar));
+
+    // insert one valid row
+    List<Map<String, Object>> validRows = new ArrayList<>();
+    validRows.add(Collections.singletonMap("colChar", "a"));
+
+    InsertValidationResponse response = innerBufferOnErrorContinue.insertRows(validRows, "1");
+    Assert.assertFalse(response.hasErrors());
+    response = innerBufferOnErrorAbort.insertRows(validRows, "1");
+    Assert.assertFalse(response.hasErrors());
+
+    // insert one valid and one invalid row
+    List<Map<String, Object>> mixedRows = new ArrayList<>();
+    mixedRows.add(Collections.singletonMap("colChar", "b"));
+    mixedRows.add(Collections.singletonMap("colChar", "1111111111111111111111")); // too big
+
+    response = innerBufferOnErrorContinue.insertRows(mixedRows, "3");
+    Assert.assertTrue(response.hasErrors());
+
+    Assert.assertThrows(
+        SFException.class, () -> innerBufferOnErrorAbort.insertRows(mixedRows, "3"));
+
+    if (bdecVersion == Constants.BdecVersion.THREE) {
```

Comment:
What about the other possible versions in future?

I would at least use a switch/case and throw on unimplemented versions here to not loose coverage for new versions.

Long term if we have to maintain more versions, we might want to implement API for snapshot deserialiazation into List<List<Object>> for assertions like this or create and compare good/bad snapshots.

Reply:
ack, I changed it to use a switch/case with throw exception in `default:`.